api/src/main/java/org/pmiops/workbench/api/BugReportController.java

```diff
@@ -107,23 +73,40 @@ void setMailServiceProvider(Provider<MailService> mailServiceProvider) {
                   String.format("Jupyter returned null content for '%s', continuing", fileName));
               continue;
             }
-            MimeBodyPart attachPart = new MimeBodyPart();
-            attachPart.setDataHandler(new DataHandler(logContent, "text/plain"));
-            attachPart.setFileName(fileName);
-            multipart.addBodyPart(attachPart);
+            File tempLogFile = createTempFile(fileName, logContent);
+            jiraService.uploadAttachment(issueKey, tempLogFile);
+            tempLogFile.delete();
```

Comment:
If we encounter an exception after uploading the attachment, the delete call will never be executed. This should go into a `finally` block so we're guaranteed there are no lasting side effects.